src/lib/ip.rs

```diff
@@ -61,6 +62,37 @@ pub struct IpConf {
     pub addresses: Vec<IpAddrConf>,
 }
 
+impl From<&Ipv4Info> for IpConf {
+    fn from(info: &Ipv4Info) -> Self {
+        let mut addrs = Vec::new();
+        for addr_info in &info.addresses {
+            if addr_info.valid_lft == "forever" {
+                addrs.push(IpAddrConf {
+                    address: addr_info.address.clone(),
+                    prefix_len: addr_info.prefix_len,
+                });
+            }
+        }
+        Self { addresses: addrs }
+    }
+}
+
+impl From<&Ipv6Info> for IpConf {
+    fn from(info: &Ipv6Info) -> Self {
+        let mut addrs = Vec::new();
+        for addr_info in &info.addresses {
+            if addr_info.valid_lft == "forever" {
+                addrs.push(IpAddrConf {
+                    address: addr_info.address.clone(),
+                    prefix_len: addr_info.prefix_len,
+                });
+            }
+        }
+        Self { addresses: addrs }
+    }
+}
+
```

Comment:
Can we make a ipv4/ipv6 generic function for this ?

Reply:
They are taking different type of input. Maybe [`Box<dyn Trait>`][1] might help, but I don't like to include complexity which is unfriendly to me and other new rust learner. If you have suggestions, please kindly let me know.


[1]: https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/dyn-trait-for-trait-objects.html